jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/RequestTest.java

```diff
@@ -334,62 +336,50 @@ protected void service(HttpServletRequest request, HttpServletResponse resp) thr
                 ServletContextRequest servletContextRequest = ServletContextRequest.getServletContextRequest(request);
                 Request coreRequest = servletContextRequest.getRequest();
 
+
                 // Set some fake SSL attributes
-                Object certificate = new Object();
-                coreRequest.setAttribute(SecureRequestCustomizer.CIPHER_SUITE_ATTRIBUTE, "quantumKnowledge");
-                coreRequest.setAttribute(SecureRequestCustomizer.KEY_SIZE_ATTRIBUTE, 42);
-                coreRequest.setAttribute(SecureRequestCustomizer.SSL_SESSION_ID_ATTRIBUTE, "identity");
-                coreRequest.setAttribute(SecureRequestCustomizer.PEER_CERTIFICATES_ATTRIBUTE, certificate);
+                X509Certificate[] certificate = new X509Certificate[0];
+                coreRequest.setAttribute(SecureRequestCustomizer.SSL_SESSION_DATA_ATTRIBUTE,
+                    new EndPoint.SslSessionData(null, "identity", "quantum_42_Knowledge", certificate));
 
                 // Check we have all the attribute names in servlet API
                 Set<String> names = new HashSet<>(Collections.list(request.getAttributeNames()));
                 assertThat(names, containsInAnyOrder(
-                    SecureRequestCustomizer.CIPHER_SUITE_ATTRIBUTE,
-                    "jakarta.servlet.request.cipher_suite",
-                    SecureRequestCustomizer.KEY_SIZE_ATTRIBUTE,
-                    "jakarta.servlet.request.key_size",
-                    SecureRequestCustomizer.SSL_SESSION_ID_ATTRIBUTE,
-                    "jakarta.servlet.request.ssl_session_id",
-                    SecureRequestCustomizer.PEER_CERTIFICATES_ATTRIBUTE,
-                    "jakarta.servlet.request.X509Certificate",
+                    SecureRequestCustomizer.SSL_SESSION_DATA_ATTRIBUTE,
+                    ServletContextRequest.SSL_CIPHER_SUITE,
+                    ServletContextRequest.SSL_KEY_SIZE,
+                    ServletContextRequest.SSL_SESSION_ID,
+                    ServletContextRequest.X_509_CERTIFICATE,
```

Comment:
As with other tests, I don't think we can break backwards compatibility.

Reply:
Are you worried about no longer supporting both `SecureRequestCustomizer.SSL_SESSION_ID_ATTRIBUTE` and `"jakarta.servlet.request.ssl_session_id"`??

We are kind of doomed to make a breaking change in 12 since from in 10 and 11 the `SecureRequestCustomizer` was setting the servlet specific name with "javax" or "jakarta".   So it was only since 12.0.0 we have had a neutral version.   I don't think it is so bad to replace those individual ones with a single attribute now..... If somebody complains we can put them back.  If I put them back now, they are there forever, even if deprecated.